ui/component/wunderbar/view.jsx

```diff
@@ -99,68 +104,69 @@ class WunderBar extends React.PureComponent<Props, State> {
     // Allow copying a lbry.tv url and pasting it into the search bar
-    const includesLbryTvProd = query.includes(WEB_PROD_PREFIX);
-    const includesLbryTvLocal = query.includes(WEB_LOCAL_PREFIX);
-    const includesLbryTvDev = query.includes(WEB_DEV_PREFIX);
-    const wasCopiedFromWeb = includesLbryTvDev || includesLbryTvLocal || includesLbryTvProd;
-
-    if (wasCopiedFromWeb) {
-      if (includesLbryTvDev) {
-        query = query.slice(WEB_DEV_PREFIX.length);
-      } else if (includesLbryTvLocal) {
-        query = query.slice(WEB_LOCAL_PREFIX.length);
-      } else {
-        query = query.slice(WEB_PROD_PREFIX.length);
-      }
-
-      query = query.replace(/:/g, '#');
-
-      if (query.includes(SEARCH_PREFIX)) {
-        query = query.slice(SEARCH_PREFIX.length);
-        onSearch(query);
-        return;
-      } else {
-        query = `lbry://${query}`;
-        onSubmit(query);
-        return;
-      }
+    const { doSearch, navigateToUri, updateSearchQuery } = this.props;
+
+    const slashPosition = uri.indexOf('/');
+    let query = uri.slice(slashPosition);
+    query = query.replace(/:/g, '#');
+    if (query.includes(SEARCH_PREFIX)) {
+      query = query.slice(SEARCH_PREFIX.length);
+      doSearch(query);
+    } else {
+      // TODO - double check this code path
+      let path = `lbry://${query}`;
+      const uri = formatLbryUrlForWeb(path);
+      navigateToUri(uri);
+      updateSearchQuery('');
     }
+  }
 
-    // User selected a suggestion
-    if (suggestion) {
-      if (suggestion.type === SEARCH_TYPES.SEARCH) {
-        onSearch(query);
-      } else if (suggestion.type === SEARCH_TYPES.TAG) {
-        history.push(`/$/${PAGES.DISCOVER}?t=${suggestion.value}`);
-      } else if (isURIValid(query)) {
-        const uri = normalizeURI(query);
-        onSubmit(uri);
-      } else {
-        showSnackError();
-      }
-
-      return;
+  onClickSuggestion(query: string, suggestion: { value: string, type: string }): void {
+    const { navigateToUri, doSearch, doShowSnackBar } = this.props;
+
+    if (suggestion.type === SEARCH_TYPES.SEARCH) {
+      doSearch(query);
+    } else if (suggestion.type === SEARCH_TYPES.TAG) {
+      const encodedSuggestion = encodeURIComponent(suggestion.value);
+      const uri = `/$/${PAGES.DISCOVER}?t=${encodedSuggestion}`;
+      navigateToUri(uri);
+    } else if (isURIValid(query)) {
+      const uri = normalizeURI(query);
+      navigateToUri(uri);
+    } else {
+      doShowSnackBar(INVALID_URL_ERROR);
     }
+  }
+
+  onSubmitRawString(st: string): void {
+    const { navigateToUri, doSearch, doShowSnackBar } = this.props;
     // Currently no suggestion is highlighted. The user may have started
     // typing, then lost focus and came back later on the same page
     try {
-      if (isURIValid(query)) {
-        const uri = normalizeURI(query);
-        onSubmit(uri);
+      if (isURIValid(st)) {
+        const uri = normalizeURI(st);
+        navigateToUri(uri);
```

Comment:
Before this, you need to do `formatLbryUrlForWeb(uri)`

Reply:
I will make a new PR now. My bad for missing this... Thanks